addons/stock_account/models/account_move.py

```diff
@@ -188,7 +188,11 @@ def _stock_account_anglo_saxon_reconcile_valuation(self, product=False):
                     lambda line: line.product_id == product and line.account_id == product_interim_account and not line.reconciled)
 
                 # Search for anglo-saxon lines linked to the product in the stock moves.
-                product_stock_moves = stock_moves.filtered(lambda stock_move: stock_move.product_id == product)
+                bom = self.env['mrp.bom']._bom_find(product=product)
+                if bom and bom.type == 'phantom':
+                    product_stock_moves = stock_moves.filtered(lambda stock_move: stock_move.bom_id == bom)
```

Comment:
this implementation is better but still it relies on the fact the bom is not changed between the so confirmation and the so invoicing :( i'm trying to find a better solution but i don't see one at the moment,especially with the kits in kit